api/src/main/java/io/opentelemetry/trace/SpanData.java

```diff
@@ -175,6 +175,46 @@ public static SpanData create(
 
   SpanData() {}
 
+  /**
+   * An immutable implementation of the {@link io.opentelemetry.trace.Event}.
+   *
+   * @since 0.1.0
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class Event implements io.opentelemetry.trace.Event {
+    private static final Map<String, AttributeValue> EMPTY_ATTRIBUTES =
+        Collections.unmodifiableMap(Collections.<String, AttributeValue>emptyMap());
+
+    /**
+     * Returns a new {@code Event} with the given name.
+     *
+     * @param name the text name of the {@code Event}.
+     * @return a new {@code Event} with the given name.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @since 0.1.0
+     */
+    public static io.opentelemetry.trace.Event create(String name) {
+      return new AutoValue_SpanData_Event(name, EMPTY_ATTRIBUTES);
```

Comment:
check name for null?

Reply:
Autovalue generated classes will do that for us.